AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -99,17 +99,13 @@ - (AVPlayerItem *)constructPlayerItem
 {
   ASDN::MutexLocker l(_videoLock);
 
-  AVPlayerItem *playerItem = nil;
-  
-  if (_asset != nil) {
-    if (_asset.tracks.count > 0) {
-      playerItem = [[AVPlayerItem alloc] initWithAsset:_asset];
-    } else if ([_asset isKindOfClass:[AVURLAsset class]]) {
-      playerItem = [[AVPlayerItem alloc] initWithURL:((AVURLAsset *)_asset).URL];
-    }
+  if ([_asset isKindOfClass:[AVURLAsset class]]) {
+    return [[AVPlayerItem alloc] initWithURL:((AVURLAsset *)_asset).URL];
+  } else if (_asset != nil) {
+    return [[AVPlayerItem alloc] initWithAsset:_asset];
   }
 
-  return playerItem;
+  return nil;
```

Comment:
Although I am quite happy with the style to eliminate the local variable here, in this kind of formatting, I would prefer that we have an unconditional else clause at the end that includes the fall through return. It makes it more clear that we expect early returns in all cases. Certainly not an issue for this change, and no need for a change just for this, just an offhanded style musing!